crates/wasm-mutate/src/mutators/peephole/eggsy/lang.rs

```diff
@@ -878,9 +872,80 @@ impl Lang {
     }
 }
 
+// To match memory like nodes by its inmediates
+macro_rules! match_mem {
+    ($l:ident, $self: ident, $other: ident) => {
+        if let (
+            Lang::$l {
+                static_offset,
+                align,
+                mem,
+                ..
+            },
+            Lang::$l {
+                static_offset: static_offset2,
+                align: align2,
+                mem: mem2,
+                ..
+            },
+        ) = ($self, $other)
+        {
+            return static_offset == static_offset2
+                && align == align2
+                && mem == mem2;
+        }
+    };
+}
+
 impl egg::Language for Lang {
     fn matches(&self, other: &Self) -> bool {
-        ::std::mem::discriminant(self) == ::std::mem::discriminant(other)
+        match_mem!(I32Load, self, other);
+        match_mem!(I64Load, self, other);
+        match_mem!(F32Load, self, other);
+        match_mem!(F64Load, self, other);
+        match_mem!(I32Load8S, self, other);
+        match_mem!(I32Load8U, self, other);
+        match_mem!(I32Load16S, self, other);
+        match_mem!(I32Load16U, self, other);
+        match_mem!(I64Load8S, self, other);
+        match_mem!(I64Load8U, self, other);
+        match_mem!(I64Load16S, self, other);
+        match_mem!(I64Load16U, self, other);
+        match_mem!(I64Load32S, self, other);
+        match_mem!(I64Load32U, self, other);
+        match_mem!(I32Store, self, other);
+        match_mem!(I64Store, self, other);
+        match_mem!(F32Store, self, other);
+        match_mem!(F64Store, self, other);
+        match_mem!(I32Store8, self, other);
+        match_mem!(I32Store16, self, other);
+        match_mem!(I64Store8, self, other);
+        match_mem!(I64Store16, self, other);
+        match_mem!(I64Store32, self, other);
+
+        match (self, other) {
+            (Lang::I32(v), Lang::I32(v2)) => { v == v2
+            }
+            (Lang::I64(v), Lang::I64(v2)) => { v == v2
+            }
+            (Lang::F32(v), Lang::F32(v2)) => { v == v2
+            }
+            (Lang::F64(v), Lang::F64(v2)) => { v == v2
+            }
+            (Lang::GlobalGet(v), Lang::GlobalGet(v2)) => {v == v2
+            }
+            (Lang::LocalGet(v), Lang::LocalGet(v2)) => {v == v2
+            }
```

Comment:
This looks like it could use a `rustfmt`

Reply:
Right...forgot to make a pass 🤦